lessons/basics/mix-tasks.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: page
+title: Custom Mix Tasks 
+category: basics
+order: 15
+lang: en
+---
+
+Creating custom mix tasks for your Elixir projects.
+
+## Table of Contents
+
+- [Introduction](#introduction)
+- [Setup](#setup)
+- [Custom Mix Task](#custom-mix-task)
+- [Mix Tasks in Action](#mix-tasks-in-action)
+
+## Introduction 
+
+It's not uncommon to want to extend your mix applications functionality by adding custom mix tasks. Before we learn about how to create specific mix tasks for our projects, let's look at a couple that already exist.
+
+```shell
+$ mix phoenix.new my_phoenix_app
+
+* creating my_phoenix_app/config/config.exs
+* creating my_phoenix_app/config/dev.exs
+* creating my_phoenix_app/config/prod.exs
+* creating my_phoenix_app/config/prod.secret.exs
+* creating my_phoenix_app/config/test.exs
+* creating my_phoenix_app/lib/my_phoenix_app.ex
+* creating my_phoenix_app/lib/my_phoenix_app/endpoint.ex
+* creating my_phoenix_app/test/views/error_view_test.exs
+...
+```
+
+As we can see from the shell command above, phoenix has a mix task to generate a new project. What if we could create our own for our project? Well the great news is we can, and Elixir makes that very easy for us.
+
+## Setup
+
+```shell
+$ mix new hello
+
+* creating README.md
+* creating .gitignore
+* creating mix.exs
+* creating config
+* creating config/config.exs
+* creating lib
+* creating lib/hello.ex
+* creating test
+* creating test/test_helper.exs
+* creating test/hello_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+cd hello
+mix test
+
+Run "mix help" for more commands.
+```
+
+Now, in the **lib/hello.ex** file that Mix generated for us, let's create a simple function that will output "Hello, World!"
+
+```elixir
+defmodule Hello do
+
+  @doc """
+  Output's `Hello, World!` everytime.
+  """
+  def say do
+    IO.puts "Hello, World!"
+  end
+end
+```
+
+## Custom Mix Task
+
+Now with lib create a new directory and file **hello/lib/mix/tasks/hello.ex**. Within this file, let's insert these 7 lines of Elixir.
+
+```elixir
+defmodule Mix.Tasks.Hello do
+  use Mix.Task
+
```

Comment:
For tasks to show in `mix help` we need to add something like 
```
@shortdoc "Simply calls the Hello.say/0 function"
```
So when we do `mix help` we get 
```
mix hello                 # Simply calls the Hello.say/0 function
```